### PR TITLE
Adds npm prune and npm install (dev deps) to app build process

### PR DESCRIPTION
**Pull request by RobinBol**
- Adds npm install --save-dev `homey` to app create process.
- Adds npm prune to the app run/publish process. The output from the npm prune dry-run will be used to ignore the dev dependencies while creating a tarball that will be uploaded to a Homey and/or published.

### CODE REVIEW

File: lib/App/index.js [hunk 1]
```diff
@@ -1944,6 +1944,14 @@ class App {
 
     await writeFileAsync( path.join(appPath, 'package.json'), JSON.stringify(packageJson, false, 2) );
 
+    Log(colors.green(`✓ Installing npm packages...`));
```
Comment (KyleAthom) [inline lib/App/index.js]: We might want to check if npm is actually available.

File: lib/App/index.js [hunk 2]
```diff
@@ -1944,6 +1944,14 @@ class App {
...
+
+    // Install homey as dep dependency
```
Comment (KyleAthom) [inline lib/App/index.js]: Dev

File: lib/App/index.js (hunk 1 above)
Comment (RobinBol) [reply lib/App/index.js]: Fixed!

File: lib/App/index.js (hunk 2 above)
Comment (RobinBol) [reply lib/App/index.js]: Fixed!

File: lib/Modules/NpmCommands.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+const colors = require('colors');
+const npm = require('npm-programmatic');
+const exec = util.promisify(require('child_process').exec);
+
+const Log = require('../..').Log;
+
+class NpmCommands {
+
+  /**
+   * Check if git is installed. By using --version git will not exit with an error code
```
Comment (KyleAthom) [inline lib/Modules/NpmCommands.js]: Git or npm?

Comment (RobinBol) [reply lib/Modules/NpmCommands.js]: Haha, shameless copy paste, thanks!

**Pull request merged**